### PR TITLE
Expose ParameterInfo for Minimal Actions API Explorer

### LINKED ISSUE #36438: Cannot inspect parameter attributes in API Explorer with Minimal Actions
### Describe the bug

Attributes of parameters of Minimal Actions cannot be inspected in API Explorer as the `ParameterDescriptor` property of the `ApiParameterDescription` for the parameter is always `null`. Further, if it were not null, the type of the value of the property would need to implement `IParameterInfoParameterDescriptor` so that the relevant `ParameterInfo` value could be accessible for inspection.

https://github.com/dotnet/aspnetcore/blob/e6a5af6633f08890afbd99a6ca7687c623454768/src/Mvc/Mvc.ApiExplorer/src/EndpointMetadataApiDescriptionProvider.cs#L166-L174

### To Reproduce

The following unit test demonstrates the issue, it fails due to `nameParam.ParameterDescription` being null.

```csharp
[Fact]
public void TestParameterAttributesCanBeInspected()
{
    var apiDescription = GetApiDescription(([Description("The name.")] string name) => { });
    Assert.Equal(1, apiDescription.ParameterDescriptions.Count);

    var nameParam = apiDescription.ParameterDescriptions[0];
    Assert.Equal(typeof(string), nameParam.Type);
    Assert.Equal(typeof(string), nameParam.ModelMetadata.ModelType);
    Assert.Equal(BindingSource.Query, nameParam.Source);
    Assert.False(nameParam.IsRequired);

    Assert.NotNull(nameParam.ParameterDescriptor); // Assertion fails here

    Assert.Equal("name", nameParam.ParameterDescriptor.Name);
    Assert.Equal(typeof(string), nameParam.ParameterDescriptor.ParameterType);

    var descriptor = Assert.IsAssignableFrom<IParameterInfoParameterDescriptor>(nameParam.ParameterDescriptor);

    Assert.NotNull(descriptor.ParameterInfo);

    var description = Assert.Single(descriptor.ParameterInfo.GetCustomAttributes<DescriptionAttribute>());

    Assert.NotNull(description);
    Assert.Equal("The name.", description.Description);
}
```

### Further technical details

Tested as of commit `b33b4d4453d93be12a847a5e0c222d86edd706a0`.

### PR DESCRIPTION
**Pull request by martincostello**
**PR Title**

Expose ParameterInfo for Minimal Actions API Explorer

**PR Description**

Populate the `ParameterDescriptor` for Minimal Actions parameters with a type that implements `IParameterInfoParameterDescriptor` so that any custom attributes on the parameter can be inspected.

Relates to #36438

### CODE REVIEW

File: src/Mvc/Mvc.ApiExplorer/src/EndpointParameterDescriptor.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+
+namespace Microsoft.AspNetCore.Mvc.ApiExplorer;
```
Comment (captainsafia) [inline src/Mvc/Mvc.ApiExplorer/src/EndpointParameterDescriptor.cs:8]: Nit: we haven't standardized on file-scoped namespaces in the repo yet so we're avoiding adding it for the moment.

Yes, we did sneak some in in the new analyzers project, but 🤪

**Review by captainsafia — APPROVED** [review]
Thanks! This will be helpful for https://github.com/dotnet/aspnetcore/issues/36525.

File: src/Mvc/Mvc.ApiExplorer/src/EndpointParameterDescriptor.cs (hunk 1 above)
Comment (martincostello) [reply src/Mvc/Mvc.ApiExplorer/src/EndpointParameterDescriptor.cs:8]: 😆 - no worries, I'll change it to the other style.

**Pull request merged**